### ISSUE TITLE
Xcode 8.1 support

### ISSUE THREAD

**Issue by heyzooi** [travis-ci/travis-ci#6791]
Please add Xcode 8.1 support

`osx_image: xcode8.1`

**Comment by seivan** [comment]
One of the reasons we pay for Travis is because we're expecting you to be ahead of the curve. However, I am not sure that's unreasonable request.

**Comment by bielikb** [comment]
Travis is always behind :/

**Comment by joshk** [comment]
Thanks for the feedback.

We usually aim for a 24-48 hour turn around for new Xcode versions, but due to some team members being out of the office last week, this has been a little delayed.

We will look into this request later this week and see what we can do.

Have an awesome day!

**Comment by stefanhp** [comment]
It would also be nice to have it on macOS 10.12 (Sierra).

**Comment by NachoSoto** [comment]
Any updates on this?

**Comment by BanzaiMan** [comment]
Sorry for the delay. We are working on it.

**Comment by bielikb** [comment]
we already resigned to wait for this update as we're in the release week of our product, thus running important stuff using other services. hopefully Travis will be on track with latest Xcode update next time

**Comment by rikvdbrule** [comment]
Please update soon, Xcode 8.1 fixes a compiler issue that is critical to distribute our app.

**Comment by oleksandr-yefremov** [comment]
> heyzooi opened this Issue 14 days ago

>  BanzaiMan commented 7 days ago 
> Sorry for the delay. We are working on it.

C'mon guys! If you have some reasonable delays, please communicate it ASAP. We rely on availability of current stable Apple APIs. Yes, Apple is rolling out language and tools versions way too often, but if you can't keep up we need to have a backup plan (Jenkins in a basement). Right now I personally can't prepare for release and have to either introduce Jenkins and set it up from scratch or wait until current issue is unblocked 😞

**Comment by brianmichel** [comment]
Any movement here? Would love to use the same build environment in Travis that we are on our locals.

**Comment by solarce** [comment]
Firstly, I want to apologize for both the delay in having support for Xcode 8.1 and the lack of communication for the last few days.

Getting support for Xcode 8.1 has been delayed while we've been working to get macOS 10.12 setup. This process has required us to build a new base OS image from scratch and unfortunately, we've run into more complications with getting 10.12 ready than we expected.

However, we're in the final stages of integration testing and hope to have the image ready for testing by Mon/Tue next week. Right now we think we'll initially promote it with a separate `osx_image:` value at first and ask folks who are watching this thread to try it out, before we promote it to replace `xcode8`

Again, I am sorry for the delays and lack of sufficient updates along the way. I'll post an update on where we are by the EOD, Mon, Nov 14th.

**Comment by solarce** [comment]
We have a 10.12/Xcode 8.1 image ready for some testing, you can try it out with `osx_image: xcode8.1sneakpeek`. Please let us know how it works for you. We'll determine our promotion plan to make it the default `xcode8` image early next week

**Comment by aaronbrethorst** [comment]
https://travis-ci.org/OneBusAway/onebusaway-iphone/builds/175201230

```
2016-11-11 22:19:25.674 xcodebuild[1566:5939] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Canceling tests due to timeout in Checking test manager availability... If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/org.onebusaway.iphone-gyziilhhwnahyrdtswtcuvzaxxtr/Logs/Test/D8FC026B-1E9C-4F43-B9FE-0F2D988C43FD/Session-OneBusAwayTests-2016-11-11_221720-HjdFR7.log" UserInfo={NSLocalizedDescription=Canceling tests due to timeout in Checking test manager availability... If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/org.onebusaway.iphone-gyziilhhwnahyrdtswtcuvzaxxtr/Logs/Test/D8FC026B-1E9C-4F43-B9FE-0F2D988C43FD/Session-OneBusAwayTests-2016-11-11_221720-HjdFR7.log}
```

**Comment by andybfp** [comment]
Config includes:
osx_image: xcode8.1sneakpeek
"rvm": "2.2.2",

Build Log results:
$ rvm use 2.2.2 --install --binary --fuzzy
ruby-2.2.2 is not installed - installing.
Searching for binary rubies, this might take some time.
Requested binary installation but no rubies are available to download, consider skipping --binary flag.
Gemset '' does not exist, 'rvm ruby-2.2.2 do rvm gemset create ' first, or append '--create'.
The command "rvm use 2.2.2 --install --binary --fuzzy" failed and exited with 2 during .

**Comment by seivan** [comment]
Out of curiosity what's the point of the "sneakpeek" suffix,  it seems redundant given the fact that it isn't a documented feature and forces people to hit back the .yml file for modification as soon as the "sneakpeek" is over. 

I had the same issue as @andybfp 

```
16.19s$ rvm use 2.3.1 --install --binary --fuzzy
ruby-2.3.1 is not installed - installing.
Searching for binary rubies, this might take some time.
Requested binary installation but no rubies are available to download, consider skipping --binary flag.
Gemset '' does not exist, 'rvm ruby-2.3.1 do rvm gemset create ' first, or append '--create'.
The command "rvm use 2.3.1 --install --binary --fuzzy" failed and exited with 2 during .
```

**Comment by solarce** [comment]
Once we are happy with this 8.1 image it will replace our `xcode8` image.
We do not plan to support 8.0 and 8.1 side by side.

The `rvm` thing is due to a lack of a prebuilt binary version of 2.2.2 for
10.12. We can look into publishing one next week

-- 
Brandon Burton

**Comment by stefanhp** [comment]
This 10.12/Xcode 8.1 image works for me. 
👍

**Comment by Upliver** [comment]
https://travis-ci.org/Upliver/FVSeatsPicker/jobs/175533251

**Comment by cepages** [comment]
@stefanhp are you using `osx_image: xcode8.1sneakpeek` or `osx_image: xcode8`

**Comment by bielikb** [comment]
using `osx_image: xcode8.1sneakpeek` our build is stalled. It cannot get through `Crashlytics/Fabric` script

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

**Comment by rikvdbrule** [comment]
We are running into a similar issue on `xcode8.1sneakpeek`, our build stalls on the “Embed Pods Framework” step of Cocoapods.

**Comment by nsuarezml** [comment]
@bielikb I have the same issue. I think it happen because the changes in the [CocoaPods specs repo](http://blog.cocoapods.org/Sharding/). 

As a workaround we can use the command `travis_wait`, but the final solution is that Travis update the Specs in the OSX images.

**Comment by stefanhp** [comment]
@cepages I'm using ```xcode8.1sneakpeek```. ```xcode8``` no longer works for us due to bug in Foundation that was fixed in Xcode 8.1.

**Comment by sirnacnud** [comment]
The `xcode8.1sneakspeak` image worked fine, no problems!

**Comment by BanzaiMan** [comment]
@sirnacnud Did you mean `xcode8.1sneakpeek`?

**Comment by sirnacnud** [comment]
Yeah sorry, typo on my end.

**Comment by solarce** [comment]
The stalls some of you saw were due to [a big change Cocoapods made to their repos](http://blog.cocoapods.org/Sharding/), so `pod setup` took a while.

We've [published an update Xcode 8.1 image](https://blog.travis-ci.com/2016-11-15-xcode-81-is-here) which was updated earlier today and includes the Cocoapods changes.

After further discussion internally and based in feedback from users, we've decided to support Xcode 8.1 as a separate image for now, usable with `xcode8.1` and we'll continue to maintain it and `xcode8.0` as separate images until next year.

We'll be publishing some more detailed information on upcoming changes to our macOS image library in the next couple of days.

As always, please let me know if you have any questions

**Comment by solarce** [comment]
Regarding the previous issues with `rvm` and 2.2.2, we're building for 10.12 and publishing them to http://rubies.travis-ci.org/, so `rvm use 2.2.2 --install ...` should find it and use the binary package now.

**Comment by solarce** [comment]
Also, we've made it so that `osx_image: xcode8.1sneakpeek` routes to the updated 8.1 image. We'll leave this in place until Next Tue, Nov 22.

If you were using the `xcode8.1sneakpeek` value, please update your config with `xcode8.1` as soon as you're able, thanks!

**Comment by michaeldorner** [comment]
Unfortunately, the `xcode8.1` does not work for me ([log](/Users/travis/Library/Developer/Xcode/DerivedData/ByteBackpacker-biaoqfkjirastddswpvhuchcyfhv/Logs/Test/C1276747-C60A-4529-9EFE-6AD3D02CB5F6/Session-ByteBackpacker iOS Tests-2016-11-16_091919-uDNS1r.log))

**Comment by bielikb** [comment]
We've tried `xcode8.1` however build is still stalled (running the same xcode build script locally works just flawlessly). We don't use any CocoaPods, however we use embedded frameworks. Is it possible. there's some issue with it?

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

**Comment by epankratov** [comment]
We have the same problem as @bielikb with our setup too: the build gets stalled while code signing

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
> The build has been terminated

**Comment by cotsog** [comment]
@epankratov, @bielikb: can you post (or send us an e-mail to support [at] travis-ci [dot] com) the link/URL to build log showing this message so we can have a look? Thank you!

**Comment by solarce** [comment]
If you're seeing your build time out during code signing, please try pre-pending your command with [`travis_wait`](https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received), e.g. `travis_wait 30 xcodebuild ...`

**Comment by epankratov** [comment]
@cotsog I've sent mine. It hangs on "Processing Info.plist" step.

**Comment by cotsog** [comment]
We've found out that some `xcodebuild` steps can prompt the following dialog on our `xcode8.1` image:

![](https://d33v4339jhl8k0.cloudfront.net/inline/3863/abac1b9b0a6a88fa310f1e47c21ff1795211becf/754743dc15c91a336213a55b5e42dfe0ad2f6deb/Screen-Shot-2016-11-16-at-14-57-14.png)

This happens even though `/usr/bin/codesign` is safelisted when importing the key e.g.

`security import ./scripts/certs/dist.cer -k ~/Library/Keychains/ios-build.keychain -T /usr/bin/codesign`

We are not sure why this doesn't work anymore on macOS Sierra. Can some of you try to change your `security import` commands for the following?

`security import ./scripts/certs/dist.cer -k ~/Library/Keychains/ios-build.keychain -A`

The "-A" option should allow all applications to use the keys/certificates. Please take care of adding the "-A" to all your `security import` commands.

Thanks in advance for sending us the result you get.

**Comment by cotsog** [comment]
I did some research and stumbled on the following [StackOverflow answer](http://stackoverflow.com/a/40039594/213272) and [the accompanying radar bug](http://www.openradar.me/28524119). It seems that you now have to add a new (undocumented) command after importing your keys/certificates in the keychain i.e.

`security set-key-partition-list -S apple-tool:,apple: -s -k keychainPass keychainName`

Can you try this out and report if it helps in this issue?

Thank you again for your collaboration.

**Comment by cepages** [comment]
same problem than @michaeldorner and @bielikb 

```
2016-11-17 10:12:37.987 xcodebuild[2089:7417] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Canceling tests due to timeout in Checking test manager availability... If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/fendixMR-ancyhmpvswblcwhldjlfhfjghrzo/Logs/Test/6719B721-4929-4009-B0F3-B9AC69053CF4/Session-fendixMRTests-2016-11-17_101034-Ls3Pke.log" UserInfo={NSLocalizedDescription=Canceling tests due to timeout in Checking test manager availability...
```

**Comment by bielikb** [comment]
@cepages @epankratov following @cotsog 's instructions from yesteday helped us ;)

a) change argument for all your security imports to keychain from `-T` to `-A`
`security import ./scripts/certs/dist.cer -k ~/Library/Keychains/ios-build.keychain -A` 
b) after all imports execute command `set-key-partition-list`
`security set-key-partition-list -S apple-tool:,apple: -s -k keychainPass keychainName`

where 
- **keychainPass** is your password for the keychain you've created
- ** keychainName** is name of your keychain, where you've imported all your certificates/private key(s)

**Comment by cepages** [comment]
@bielikb I was not used any steps with certificates in travis before, does this mean I have to do this in order to make travis work?

**Comment by bielikb** [comment]
@cepages are you working on some open source project/framework, where you don't have to codesign?
 
we've setup our Travis few years ago, following this article: 
https://www.objc.io/issues/6-build-tools/travis-ci/

hope this will help you. if not, just ask. maybe i'll be able to help anyway

**Comment by cepages** [comment]
Thanks @bielikb, we use private projects in Travis. 

We are using code sign, in fact we are using fastlane to cover codesign, but I've never had to set up the part of `App Signing` in that manual ( btw great manual, I used years ago too :) it worked without that section, I assumed tests are run in a simulator and XCode in travis could run with a wildcard provisioning profile.

I'm aware that in XCode 8 they changed the signing process in XCode, but I could make travis work with XCode 8 without adding codesign.

**Comment by epankratov** [comment]
@bielikb We are using Fastlane, and I'm creating keychains/importing certs via their utility, so it seems too me, I have to report them a bug.

**Comment by devdrey** [comment]
@epankratov https://github.com/fastlane/fastlane/issues/7046
There is also a PR. https://github.com/fastlane/fastlane/pull/7048

**Comment by cepages** [comment]
@bielikb after a bit of pain I got it using fastlane.

Just in case anyone need it:

In the travis.yml:

```
script:
  5 - pod install
  4 - fastlane run_test
```

In the fastlane file:

```
desc "Test app the app"
  lane :run_test do
  match(type: "adhoc", app_identifier: "you app identifier", git_url: "your certificate git repo", username: "your username for the certificate repo", readonly: true) # more information https://codesigning.guide
   scan(workspace: "your work space", scheme: "your scheme", devices: ["devices to test"], clean: true)
   end
```

And set the environment variables:

`FASTLANE_PASSWORD= this is the password for the user in the certificate repo`
`MATCH_PASSWORD= this is the passphrase for the certificate repo`

**Comment by cepages** [comment]
@devdrey could you make it work with the work around in fastlane/fastlane#7046 ?

**Comment by devdrey** [comment]
I didn't tried it. I just stumbled upon this issue I'm the fastlane repo and thought it could be related.
Actually I don't use fastlane on travis jet.

**Comment by markohlebar** [comment]
@cotsog @bielikb I was pulling my hair out today, @bielikb's suggestion works like a charm. Thanks guys!

**Comment by vegidio-symbio** [comment]
My builds are still stalling at "Embed Pods Frameworks", using image `xcode8.1`. Does anyone have a solution for this?

Thanks

**Comment by rakaramos** [comment]
@vegidio-symbio same problem here.
`security set-key-partition-list` command did not worked for me. Maybe I'm missing something here...

**Comment by cotsog** [comment]
@vegidio-symbio, @rakaramos: Sorry for the troubles. Can you send an e-mail to support [at] travis-ci [dot] com with the name of your repository and a URL to the a build log on travis-ci.org/travis-ci.com showing your latest attempt?

Thank you for your patience.

**Comment by bielikb** [comment]
@cotsog from time to time I can see our builds are stalled (we set import rules for our certs one week ago). Maybe `security set-key-partition-list` doesn't work every time. I can see that also other people like @rakaramos  @vegidio-symbio have probably same problem. I guess this problem needs more investigation.

**Comment by sbandol** [comment]
My builds are also stalling at "Embed Pods Frameworks".
![screen](http://i.imgur.com/8Pc0sGE.jpg)
Tried with no luck:
- keychain workaround: https://github.com/travis-ci/travis-ci/issues/6791#issuecomment-261215038
- update cocoapods to 1.2.0 beta 1 and also to last version in the GitHub
- update fastlane to 1.110.0 and also to the last version in GitHub (including a modified version with keychain workaround)
- used different versions of ruby

**Comment by cepages** [comment]
@sbandol have you tried travis wait? https://docs.travis-ci.com/user/common-build-problems/

**Comment by sbandol** [comment]
@cepages yep, `travis_wait`, `travis_wait 30`, and `travis_wait 60`. No luck

**Comment by cepages** [comment]
I assume that you check that travis_wait 30 (for example ) is working fine, it's very tricky sometimes and sometimes it's easier to add the command in a script and do: `travis_wait 30 ./myScript.sh`

**Comment by bielikb** [comment]
@cepages @sbandol I've observed, that some of those stalled builds where running only for cca 10-11 minutes. we are running xcodebuild to build our project. it looks like Travis might sometimes falsly indicate stalled builds. but that just my guess. other builds that take approx 15-20 minutes to build are passing fine

**Comment by rakaramos** [comment]
@cotsog any news on this issue? I still can't build against Xcode 8.1

**Comment by cepages** [comment]
@cotsog for me works with fastlane ( see my comment 11 days ago ) except that sometimes we need to rebuild the test, but that's a problem with fastlane :)

**Comment by cotsog** [comment]
@rakaramos: it seems an e-mail I sent you last Friday was sent to @epankratov instead. I've just resent it. Can you confirm if you've received it?

@cepages: Glad to hear it! Can you tell me if you are signing your iOS app in your build?

**Comment by HazAT** [comment]
I just wanted to join the conversation here ...

We are still having troubles completing our build, its stuck in the signing phase ...
We tried everything described earlier in this thread (changing the `security` commands) but its still stuck.
Right now its not a major issue because the build works without any issues on `xcode8`.
I also talked with @cotsog via mail already about this ;)

I am curious if its just us ...

**Comment by rakaramos** [comment]
@HazAT we are having issues here because Xcode 8 has issues with storyboard/xib compilations.
E.g: Building on Xcode 8, we have a dark overlay being added to the screen. On Xcode 8.1, the issue is not there (Weird right?)

**Comment by sbandol** [comment]
We also have issues with Xcode8 image (mostly UI as @rakaramos says).
Xcode8.1 image still not working for us, so we are building manually for now.

**Comment by cepages** [comment]
@cotsog yeah, the signing is made by fastlane/match

**Comment by rakaramos** [comment]
Guys, just figured that out. It seems that we were missing the `codesign` application id

`security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k keychainPass keychainName`

I'm using `keychainName.keychain` (adding .keychain)

**Comment by zzzel** [comment]
Can confirm that the solution from @rakaramos works, just successfully deployed a new app update.

**Comment by vegidio** [comment]
@rakaramos Are there any specific values for `keychainPass` and `keychainPass`? I'm using Fastlane to sign my application so I'm not sure what values I should use there.

**Comment by zzzel** [comment]
@vegidio Use the same values as for when you create a keychain: `security create-keychain -p [keychainPass] [keychainName]`. `set-key-partition-list` works without the `-p` though.

**Comment by HazAT** [comment]
For everyone still having troubles signing the app with fastlane / match
@cotsog Pointed out that you apparently have to create a keychain with password in order to make the signing with match work

Adding this to the `Fastfile` did the trick
```
create_keychain(
      name: ENV["MATCH_KEYCHAIN_NAME"],
      password: ENV["MATCH_KEYCHAIN_PASSWORD"],
      timeout: 3600
)
```

Thx @cotsog 🎉

**Comment by Rusik** [comment]
Thanks @rakaramos and @bielikb! Everything working now for me

**Comment by Sega-Zero** [comment]
thx to @HazAT, got everything working - forgot to specify `MATCH_KEYCHAIN_NAME` variable, had only `KEYCHAIN_NAME` specified.
the default login.keychain seemed to be created without the password, so `security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k keychainPass keychainName` always fails. @joshk FYI

**Comment by cotsog** [comment]
@sbandol: sorry for the delay but could you send an e-mail to support [at] travis-ci [dot] com with the name of your repository and a link/URL to a build log showing your latest attempt to run your build on our Xcode 8.1 image i.e. `osx_image: xcode8.1`. Thanks!

**Comment by sbandol** [comment]
Thanks @cotsog. 
Following @HazAT recommendation got it working with:
```
    create_keychain(
      name: ENV["MATCH_KEYCHAIN_NAME"],
      password: ENV["MATCH_PASSWORD"],
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      add_to_search_list: true
    )
```


```
    match(
      type: "adhoc",
      keychain_name: ENV["MATCH_KEYCHAIN_NAME"],
      keychain_password: ENV["MATCH_PASSWORD"],
      readonly: true
    )
```

**Comment by codeOfRobin** [comment]
@sbandol it's still failing for me after including your code in my `Fastfile`.

**Comment by cotsog** [comment]
@codeOfRobin: sorry for the delay of my reply but if you are still having problems, could you send us an e-mail to support [at] travis-ci [dot] com with the name of your repository and a link/URL to a build log showing your latest attempt on travis-ci.org or travis-ci.com? Thanks!

**Comment by codeOfRobin** [comment]
@cotsog it's fixed now. Here's my Fastfile for posterity

https://gist.github.com/codeOfRobin/3d01d2ee7feabc35b3a65bd6b319f84c

**Comment by cotsog** [comment]
This is now documented in our documentation here: https://docs.travis-ci.com/user/common-build-problems/#Mac%3A-macOS-Sierra-(10.12)-Code-Signing-Errors.

We are now considering this issue closed. Please reach out at support [at] travis-ci [dot] com if you are still having issues.

Thank you for your patience!
